addons/base_vat/base_vat.py

```diff
@@ -119,6 +119,17 @@ def check_vat(self, cr, uid, ids, context=None):
                 return False
         return True
 
+    def write(self, cr, uid, ids, vals, context=None):
+        if vals.get('vat'):
+            self.validate_vat(cr, uid, ids, context=context)
+        return super(res_partner, self).write(cr, uid, ids, vals, context=context)
+
+    def create(self, cr, uid, vals, context=None):
+        new_id = super(res_partner, self).create(cr, uid, vals, context=context)
+        if vals.get('vat'):
+            self.validate_vat(cr, uid, [new_id], context=context)
+        return new_id
+
```

Comment:
no need of create and write with  validate_vat method if button is removed. constraints can check vat and validate it.